modmail/extensions/utils/error_handler.py

```diff
@@ -0,0 +1,195 @@
+            ctx.command.reset_cooldown(ctx)
+        return return_value
+
+    async def handle_user_input_error(
+        self,
+        ctx: commands.Context,
+        error: commands.UserInputError,
+        reset_cooldown: bool = True,
+    ) -> discord.Embed:
+        """Handling deferred from main error handler to handle UserInputErrors."""
+        if reset_cooldown:
+            self._reset_command_cooldown(ctx)
+        embed = None
+        msg = None
+        title = "User Input Error"
+        if isinstance(error, commands.BadUnionArgument):
+            msg = self.get_title_from_name(str(error))
+            title = self.get_title_from_name(error)
+        else:
+            title = self.get_title_from_name(error)
+        return embed or self.error_embed(msg or str(error), title=title)
+
+    async def handle_bot_missing_perms(
+        self, ctx: commands.Context, error: commands.BotMissingPermissions
+    ) -> bool:
+        """Handles bot missing permissing by dming the user if they have a permission which may be able to fix this."""  # noqa: E501
+        embed = self.error_embed(str(error))
+        try:
+            await ctx.send(embeds=[embed])
+        except discord.Forbidden:
+            logger.error(f"Unable to send an error message to {ctx.channel!s}")
+            if ANY_DEV_MODE:
+                # non-general permissions
+                perms = discord.Permissions(
+                    administrator=True,
+                    manage_threads=True,
+                    manage_roles=True,
+                    manage_channels=True,
+                )
+                if perms.value & ctx.channel.permissions_for(ctx.author).value:
+                    logger.info(
+                        f"Attempting to dm {ctx.author} since they have a permission which may be able "
+                        "to give the bot send message permissions."
+                    )
+                    try:
+                        await ctx.author.send(embeds=[embed])
+                    except discord.Forbidden:
+                        logger.notice("Also encountered an error when trying to reply in dms.")
+                        return False
+            return True
+
+    async def handle_check_failure(
+        self, ctx: commands.Context, error: commands.CheckFailure
+    ) -> typing.Optional[discord.Embed]:
+        """Handle CheckFailures seperately given that there are many of them."""
+        title = "Check Failure"
+        msg = None
+        if isinstance(error, commands.CheckAnyFailure):
+            title = self.get_title_from_name(error.checks[-1])
+            msg = str(error)
+        elif isinstance(error, commands.PrivateMessageOnly):
+            title = "DMs Only"
+        elif isinstance(error, commands.NoPrivateMessage):
+            title = "Server Only"
+        elif isinstance(error, commands.BotMissingPermissions):
+            # defer handling BotMissingPermissions to a method, since this can be problematic
```

Comment:
Add to this comment? Instead of just "Since this can be problematic" how about "handled specially for better UX and detail" or similar? We're doing good things, not just handling bad things, so that should be apparent without having to go digging into another method. It'll also direct someone looking for how the Missing Permissions DM works to the right function.